datascience_eda/datascience_eda.py

```diff
@@ -7,28 +7,82 @@
 from sklearn.decomposition import PCA
 from sklearn.metrics.pairwise import euclidean_distances
 from sklearn.preprocessing import StandardScaler
+from sklearn.impute import SimpleImputer
+from sklearn.pipeline import Pipeline, make_pipeline
 
-from yellowbrick.cluster import SilhouetteVisualizer
+from yellowbrick.cluster import KElbowVisualizer, SilhouetteVisualizer
 
-from scipy.cluster.hierarchy import (
-    average,
-    complete,
-    dendrogram,
-    fcluster,
-    single,
-    ward,
-)
+import seaborn as sns
+import matplotlib.pyplot as plt
 
 # endregion
 
-# import libraries for eda of text features
+# region import libraries for eda of text features
 import nltk
 import spacy
 from nltk.sentiment.vader import SentimentIntensityAnalyzer
 from wordcloud import WordCloud, STOPWORDS
 
 # endregion
 
+# region support functions
+
+
+def get_numeric_columns(df):
+    """get all numeric columns' names
+
+    Parameters
+    ----------
+    df : [pandas.DataFrame]
+        the dataset
+
+    Returns
+    -------
+    list
+        list of numeric column names
+    """
+    numeric_cols = df.select_dtypes("number").columns.tolist()
+    return numeric_cols
+
+
+def _verify_distance_metric(m):
+    """check if a distance metric is valid
+
+    Parameters
+    ----------
+    m : string
+        metric, should be among [‘cityblock’, ‘cosine’, ‘euclidean’, ‘l1’, ‘l2’, ‘manhattan’]
+
+    Raise
+    -------
+    Exception
+        throw an exception if the metric is invalid
+    """
+    if not m in ["cityblock", "cosine", "euclidean", "l1", "l2", "manhattan"]:
+        raise Exception(f"Invalid distance metric: {m}")
+
+
+def _verify_numeric_cols(df, num_cols):
+    """check if numeric columns are valie
```

Comment:
Could you please re check the spelling of valid in documentation 
```suggestion
    """check if numeric columns are valid
```